File: lyedu-api/src/main/java/com/lyedu/entity/UserCertificate.java
package com.lyedu.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 用户已获证书记录
 *
 * @author LyEdu Team
 */
@Data
public class UserCertificate {

    private Long id;
    private Long userId;
    private Long certificateId;
    private Long templateId;
    private String certificateNo;
    private String title;
    private LocalDateTime issuedAt;
    private LocalDateTime createTime;
}

File: lyedu-admin/src/api/userCertificate.ts
import request from '@/utils/request'

export interface UserCertificate {
  id: number
  userId: number
  realName?: string
  username?: string
  certificateId: number
  templateId: number
  certificateNo: string
  title: string
  issuedAt?: string
  createTime?: string
}

export interface PageResult<T> {
  records: T[]
  total: number
}

export const getUserCertificatePage = (params: {
  page: number
  size: number
  keyword?: string
  userId?: number
  certificateId?: number
}) => {
  return request.get<PageResult<UserCertificate>>('/user-certificate/page', { params })
}
